packages/js/src/ui/components/Notification/NotificationList.tsx

```diff
@@ -40,14 +44,52 @@ const EmptyNotificationList = () => {
 type NotificationListProps = {
   mountNotification?: NotificationMounter;
   options?: ListNotificationsArgs;
+  filter?: NotificationFilter;
 };
 /* This is also going to be exported as a separate component. Keep it pure. */
 export const NotificationList = (props: NotificationListProps) => {
   const [data, { initialLoading, setEl, end }] = useNotificationsInfiniteScroll({ options: props.options });
+  const [newNotificationsCount, setNewNotificationsCount] = createSignal(0);
+  const { t } = useLocalization();
+  const style = useStyle();
+  const filter = createMemo(() => props.filter || { read: false });
+
+  useWebSocketEvent({
+    event: 'notifications.notification_received',
+    eventHandler: async (data) => {
+      const notification = data.result;
+      if (
+        !isSameFilter(filter(), {
+          read: notification.isRead,
+          archived: notification.isArchived,
+          tags: notification.tags,
+        })
+      ) {
+        return;
+      }
+
+      setNewNotificationsCount((c) => c + 1);
+    },
+  });
 
   return (
     <Show when={!initialLoading()} fallback={<NotificationListSkeleton count={8} />}>
       <Show when={data().length > 0} fallback={<EmptyNotificationList />}>
+        <Show when={!!newNotificationsCount()}>
+          <div
+            class={style(
+              'notificationListNewNotificationsNoticeContainer',
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
💭 **thought (non-blocking):** this key is super long even hard to read